### PR TITLE
Only serve 304 when config variable set to enable it

### PR DESCRIPTION
**Pull request by samfrench**
Only serve 304 when config variable set to enable it

### CODE REVIEW

**Comment by samfrench** [comment]
@dazoakley, another broker PR for review.

File: lib/alephant/broker/response/base.rb [hunk 1]
```diff
@@ -76,6 +77,10 @@ def self.unquote_etag(etag)
           etag.to_s.gsub(/\A"|"\Z/, '')
         end
 
+        def self.allow_modified_response_status
+          Broker.config[:allow_modified_response_status] || false
```
Comment (dazoakley) [inline lib/alephant/broker/response/base.rb]: i'd maybe call the option `allow_not_modified_response_status`

File: spec/integration/not_modified_response_spec.rb [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      let(:content_type) { "application/json" }
+      let(:etag)         { '"34774567db979628363e6e865127623f"' }
+
+      before do
+        post(path, batch_json,
+          "CONTENT_TYPE"  => content_type,
+          "HTTP_IF_NONE_MATCH" => etag)
+      end
+
+      specify { expect(last_response.status).to eql 200 }
```
Comment (dazoakley) [inline spec/integration/not_modified_response_spec.rb]: Is this really correct?  If the etag matches, shouldn't it be a 304?

Comment (dazoakley) [inline spec/integration/not_modified_response_spec.rb]: ah, just noticed it's the POST request - have we reverted to always responding with 200 to POST requests?

Comment (samfrench) [inline spec/integration/not_modified_response_spec.rb]: We had this line previously:
https://github.com/BBC-News/alephant-broker/blob/master/lib/alephant/broker/response/base.rb#L65

As post requests aren't meant to be cacheable, as it's expected to have different data if it's post.

Comment (dazoakley) [inline spec/integration/not_modified_response_spec.rb]: perfect :+1:

**Comment by samfrench** [comment]
@dazoakley, I have made the variable name change you suggested.

**Comment by dazoakley** [comment]
Looks good 👍

**Pull request merged**